Easy way to extract values from foreman api search

scitechfh opened the pull request:
With puppet 3.x without future parser it difficult access foreman search results. This parser function converts easily any search result to an array.

ekohl commented on lib/puppet/parser/functions/foreman_extract_from_search.rb:
```diff
@@ -0,0 +1,35 @@
+# $hosts = foreman_extract_from_search(foreman($f),'name')
+#
+
+require "net/http"
```
Are these requires actually used?

scitechfh replied:
No. This is an copy and paste error.

ekohl commented:
Looks like it makes sense, but I'd like @mmoll or @domcleal to have a look.

domcleal commented:
I guess it's needed now because the default API version switched to v2 in Foreman 1.9, so the foreman() function now returns the pagination and results data instead of just the results.  We could change foreman() to always return results directly, but I guess that could break manifests just as badly.

ekohl commented:
@domcleal and a parameter in foreman() to return just the results?

domcleal commented:
Oh, I see this is doing something further.  Given a response like this:

    {"results": [ {"name":"foo"}, {"name":"bar"} ]}

It's extracting a key from the hashes inside the results list - a bit like a map() function.

So probably best to ignore my comment above - it may be useful, but it's separate.

scitechfh commented:
Whats about add an result filter attribute to foreman()?

domcleal commented:
That'd be fine by me too.

scitechfh commented:
I have removed my old function and added an filter function to foreman(). It's possible convert the result into an array or an hash with limited key or renamed keys.

domcleal commented:
Works great, merged as a41fce0.  Thanks @scitechfh and sorry for the delay.
